### PR TITLE
Adds mutant rat and mouse animals and samples

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Rodentkin mutants and samples"

#### Purpose of change
Adding some rat and mouse mutants for samples to drop from.

#### Describe the solution
Adds new monsters, new effects, new samples.  May if I have time this weekend go all the way up the evolution and create a special location for them. Mapgen is good enough for the moment.  I'd like to make the sewer connection room look better long term. I made a map that can spawn in a room in the TCL with some spilled alpha serum and broken cages. Like to add a mutable special with some finale rooms for this, but that portion of the PR has been moved to a new local branch after slamming my head against it for a week.

#### Describe alternatives you've considered
None

#### Testing
Got it all working.   Mapgen is good enough for the moment.  I'd like to make the sewer connection room look better long term.  Added should_not_spawn to terrains created in this PR that are only used in the mutable dungeon PR I'm hacking away at.

### CODE REVIEW

**Comment by Venera3** [comment]
They could move in some of the TCL's holding cells, it's supposed to be an animal testing only place (the mutants are just squatting there until the human testing sites make it in). Maybe a hungry, intelligent pack with a few splashes of Alpha serum on the floor?

**Comment by Maleclypse** [comment]
ratkinmicrolab
@jbytheway I don't understand why using the "microlab_ratkin" id results in none of the nested chunks loading in those OMT?

File: data/json/mapgen/microlab/microlab_generic.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   {
     "type": "mapgen",
     "method": "json",
-    "om_terrain": [ [ "microlab_generic" ] ],
+    "om_terrain": [ [ "microlab_generic" ], [ "microlab_ratkin" ] ],
```
Comment (jbytheway) [inline data/json/mapgen/microlab/microlab_generic.json:5]: You added a new OMT to this using the syntax that says you're going to define the two adjacent to each other.  But you added no new mapgen data for it, so you've defined `mapgen_ratkin` to be just a blank OMT.  I'm guessing what you wanted was for it to be identical to `microlab_generic`?  If so, you can do that with the following change:
```suggestion
    "om_terrain": [ "microlab_generic", "microlab_ratkin" ],
```

Comment (Maleclypse) [reply data/json/mapgen/microlab/microlab_generic.json]: Thank you! I don't know how long it would have taken me to realize that.

**Comment by Maleclypse** [comment]
@Venera3 If you could look over this sometime I'd appreciate it.  I also can't seem to get `"special_attacks": [ { "id": "teke_push" } ],` to fire ever and I'm not sure why.

File: data/json/effects.json [hunk 2]
```diff
@@ -1880,6 +1881,25 @@
...
+    "name": [ "puckered scratches" ],
+    "apply_message": "Your skin starts burning around some of your scratches.  Some aspirin might relieve the effects.",
+    "remove_message": "The fever breaks.",
+    "rating": "bad",
+    "resist_effects": [ "aspirin" ],
+    "base_mods": {
+      "stamina_min": [ -50 ],
+      "stamina_max": [ -100 ],
+      "stamina_chance": [ 300 ],
+      "base_mods": { "int_mod": [ -1, 0 ] },
```
Comment (kevingranade) [inline data/json/effects.json:1896]: Looks like a merge error perhaps? This change makes it valid, but double check that it matches your intent.
```suggestion
      "int_mod": [ -1, 0 ],
```

File: data/json/mapgen/lab/lab_modular/lab_nests_modular/lab_nested_mutagen.json [hunk 3]
```diff
@@ -475,6 +503,30 @@
...
   },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "//": "A nested map for mutagen research areas.  Rats for cages.",
+    "nested_mapgen_id": "lab_alpha_rat_1x1",
+    "object": {
+      "mapgensize": [ 1, 1 ],
+      "rotation": [ 0, 3 ],
+      "place_liquid": [ { "liquid": "iv_mutagen_alpha", "x": 0, "y": 0, "chance": 50 }, { "liquid": "blood", "x": 0, "y": 0, "chance": 50 } ],
```
Comment (kevingranade) [inline data/json/mapgen/lab/lab_modular/lab_nests_modular/lab_nested_mutagen.json:514]:
```suggestion
      "place_liquids": [ { "liquid": "iv_mutagen_alpha", "x": 0, "y": 0, "chance": 50 }, { "liquid": "blood", "x": 0, "y": 0, "chance": 50 } ],
```

File: data/json/mapgen/lab/lab_modular/lab_nests_modular/lab_nested_mutagen.json [hunk 4]
```diff
@@ -475,6 +503,30 @@
...
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "//": "A nested map for mutagen research areas.  Rats for cages.",
+    "nested_mapgen_id": "lab_alpha_rat_1x1",
+    "object": {
+      "mapgensize": [ 1, 1 ],
+      "rotation": [ 0, 3 ],
+      "place_liquid": [ { "liquid": "iv_mutagen_alpha", "x": 0, "y": 0, "chance": 50 }, { "liquid": "blood", "x": 0, "y": 0, "chance": 50 } ],
```
Comment (kevingranade) [inline data/json/mapgen/lab/lab_modular/lab_nests_modular/lab_nested_mutagen.json:526]:
```suggestion
      "place_liquids": [ { "liquid": "iv_mutagen_alpha", "x": 0, "y": 0, "chance": 50 }, { "liquid": "blood", "x": 0, "y": 0, "chance": 50 } ],
```

**Comment by kevingranade** [comment]
```
 ([slow] ~crafting_skill_gain)=> ../tests/overmap_test.cpp:277
([slow] ~crafting_skill_gain)=> ...............................................................................
([slow] ~crafting_skill_gain)=> 
([slow] ~crafting_skill_gain)=> ../tests/overmap_test.cpp:396: FAILED:
([slow] ~crafting_skill_gain)=>   CHECK( num_missing == 0 )
([slow] ~crafting_skill_gain)=> with expansion:
Error: ([slow] ~crafting_skill_gain)=>   3 == 0
([slow] ~crafting_skill_gain)=> with messages:
([slow] ~crafting_skill_gain)=>   num_missing := 3
([slow] ~crafting_skill_gain)=>   missing := { int_id( "microlab_ratkin_surface_connector" ), int_id(
([slow] ~crafting_skill_gain)=>   "microlab_ratkin_vent_shaft_connector" ), int_id(
([slow] ~crafting_skill_gain)=>   "microlab_sub_connector_ratkin" ) }
([slow] ~crafting_skill_gain)=>   To resolve errors about missing terrains you can either give the terrain the
([slow] ~crafting_skill_gain)=>   SHOULD_NOT_SPAWN flag (intended for terrains that should never spawn, for
([slow] ~crafting_skill_gain)=>   example test terrains or work in progress), or tweak the constraints so that
([slow] ~crafting_skill_gain)=>   the terrain can spawn more reliably, or add them to the whitelist above in
([slow] ~crafting_skill_gain)=>   this function (inteded for terrains that sometimes spawn, but cannot be
([slow] ~crafting_skill_gain)=>   expected to spawn reliably enough for this test)
```

**Comment by Maleclypse** [comment]
> ([slow] ~crafting_skill_gain)=> ../tests/overmap_test.cpp:277
> ([slow] ~crafting_skill_gain)=> ...............................................................................
> ([slow] ~crafting_skill_gain)=> 
> ([slow] ~crafting_skill_gain)=> ../tests/overmap_test.cpp:396: FAILED:
> ([slow] ~crafting_skill_gain)=>   CHECK( num_missing == 0 )
> ([slow] ~crafting_skill_gain)=> with expansion:
> Error: ([slow] ~crafting_skill_gain)=>   3 == 0
> ([slow] ~crafting_skill_gain)=> with messages:
> ([slow] ~crafting_skill_gain)=>   num_missing := 3
> ([slow] ~crafting_skill_gain)=>   missing := { int_id( "microlab_ratkin_surface_connector" ), int_id(
> ([slow] ~crafting_skill_gain)=>   "microlab_ratkin_vent_shaft_connector" ), int_id(
> ([slow] ~crafting_skill_gain)=>   "microlab_sub_connector_ratkin" ) }
> ([slow] ~crafting_skill_gain)=>   To resolve errors about missing terrains you can either give the terrain the
> ([slow] ~crafting_skill_gain)=>   SHOULD_NOT_SPAWN flag (intended for terrains that should never spawn, for
> ([slow] ~crafting_skill_gain)=>   example test terrains or work in progress), or tweak the constraints so that
> ([slow] ~crafting_skill_gain)=>   the terrain can spawn more reliably, or add them to the whitelist above in
> ([slow] ~crafting_skill_gain)=>   this function (inteded for terrains that sometimes spawn, but cannot be
> ([slow] ~crafting_skill_gain)=>   expected to spawn reliably enough for this test)

Ok I’ll move this back to draft cause I tried to move all the mutable stuff over to a different branch while I learn that.

File: data/json/itemgroups/Monsters_Animals_Lairs/harvest_dissection.json [hunk 5]
```diff
@@ -11,6 +11,15 @@
     "subtype": "collection",
     "entries": [ { "item": "rabbit_foot" } ]
   },
+  {
+    "id": "rodent_sample_small",
+    "type": "item_group",
+    "subtype": "collection",
+    "entries": [
+      { "item": "rat_sample", "count-min": 0, "count-max": 1, "prob": 15 },
```
Comment (Maleclypse) [inline data/json/itemgroups/Monsters_Animals_Lairs/harvest_dissection.json:19]:
```suggestion
      { "item": "rat_sample", "count-min": 0, "count-max": 1, "prob": 50 },
```

File: data/json/itemgroups/Monsters_Animals_Lairs/harvest_dissection.json [hunk 6]
```diff
@@ -11,6 +11,15 @@
...
+      { "item": "mouse_sample", "count-min": 0, "count-max": 1, "prob": 15 }
```
Comment (Maleclypse) [inline data/json/itemgroups/Monsters_Animals_Lairs/harvest_dissection.json:20]:
```suggestion
      { "item": "mouse_sample", "count-min": 0, "count-max": 1, "prob": 50 }
```

File: data/json/itemgroups/supplies.json [hunk 7]
```diff
@@ -351,6 +351,8 @@
       { "item": "lung", "prob": 10, "count": [ 1, 3 ] },
       { "item": "kidney", "prob": 10, "count": [ 1, 3 ] },
       { "item": "liver", "prob": 10, "count": [ 1, 3 ] },
+      { "item": "mouse_sample", "prob": 40, "count": [ 1, 3 ] },
```
Comment (Maleclypse) [inline data/json/itemgroups/supplies.json:354]:
```suggestion
      { "item": "mouse_sample", "prob": 40, "count": [ 1, 3 ] },
      { "item": "rat_sample", "prob": 40, "count": [ 1, 3 ] },
```

**Pull request merged**